### DISCUSSION TITLE
Disable SSL verification

### DISCUSSION THREAD

**Discussion by JessArschoot** [rfoltyns/log4j2-elasticsearch#78] [General]
Is there a way to disable SSL verification?

**Comment by rfoltyns** [comment]
Yes.

**Client side**

You can write your own `CertInfo` plugin and adjust the SSLContext to your needs.

Other than that - apart from adding `hostnameVarificationMode` config in the near future - I'm not planning any work in this scope and I will not accept any PRs for this.

**Server side**

It's well documented in Elasticsearch docs.

**Reply by JessArschoot** [reply]
Ok, thanks for your answer. I'm looking now to get it working with certificates. But I have one problem in the readPrivateKey.
There my keyReader.readObject is already a PrivateKeyInfo so I will get a class cast exception on line 34.
Do you have an idea how to fix this?

**Reply by JessArschoot** [reply]
I fixed it, I needed to transform my key with following statement openssl rsa -in key.key -text -out key.key

**Reply by rfoltyns** [reply]
Given the nature of what you're trying to do, it might involve a few custom steps, changes in readers, etc. SSL certs are quite strict, so current code is tailored accordingly.

Btw, current certs in test resources have expired. I'll replace them soon and maybe provide some scripts to generate your own, so it's a bit easier to cope with it.
